lib/src/chatbox.dart

```diff
@@ -52,6 +53,30 @@ class MessageActionEvent {
     message = Message.fromJson(json['message']);
 }
 
+enum UrlNavigationAction { allow, deny }
+
+class UrlNavigationRequest {
+  final Uri? url;
+  final String rawUrl;
```

Comment:
I would not expose the variables rawUrl and isValidUrl, because everything that's public needs to be supported forever, so the less we expose, the better.

Reply:
The problem is that the Uri.tryParse() method returns Null for an invalid url. If the rawUrl is not exposed there is no way for the app that uses plugin to figure out what the  original Url was for the link.  

It could be possible that a link is send with an invalid url, but in the app we still want to do something with this link. For example:
- We want to show an errormessage containing the url
- We could have a special url that opens a page in the app, maybe a url like: app://about would navigate to the app's about page. This is not a valid uri, but could be wanted behavior in a App. 

I think it would best to make the type for url a String this way the plugin stays simple and the rawUrl and isValidUrl can be removed.

```
class UrlNavigationRequest {
    final String url;
   ...
```

Reply:
Yes, let's keep things simple and use String.

Reply:
I updated the code, I also made the String Nullable to make sure it will work with any future implementation of the webview